test/functional/JIT_Test/src/jit/test/jitserver/JITServerTest.java

```diff
@@ -490,4 +543,67 @@ public void testServerComesUpAfterClientAndGoesDownAgain() throws IOException, I
 		logger.info("Stopping client...");
 		destroyAndCheckProcess(client, clientBuilder);
 	}
+
+	public void testServerSCC() throws IOException, InterruptedException {
+                logger.info("running testServerSCC: INFO and above level logging enabled");
+
+                redirectProcessOutputs(clientBuilder, "testServerSCC.client");
+                redirectProcessOutputs(serverBuilder, "testServerSCC.server");
+
+                updateJITServerPort();
+
+                final Process server = startProcess(serverBuilder, "server");
+
+                Thread.sleep(SERVER_START_WAIT_TIME_MS);
+
+                Process client = startProcess(clientBuilder, "client");
+
+                logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+                Thread.sleep(CLIENT_TEST_TIME_MS);
+
+                // Check if the SCC exists
+                if (checkCacheExists("test_jitscc")) {
+                        logger.info("Check if AOT methods exists in the cache test_jitscc::" + checkAOTCompileExists("test_jitscc"));
+
+                        logger.info("Stopping client...");
+                        destroyAndCheckProcess(client, clientBuilder);
+
+                        logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+                        Thread.sleep(CLIENT_TEST_TIME_MS);
+
+                        // Check if the SCC exists and then restart the client
+                        logger.info("Check whether the cache test_jitscc exists::" + checkCacheExists("test_jitscc"));
+                        logger.info("Restarting client...");
+                        client = startProcess(clientBuilder, "client");
+
+                        logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+                        Thread.sleep(CLIENT_TEST_TIME_MS);
+
+                        // Destroy the SCC and then restart the client
+                        logger.info("Destroy the cache test_jitscc");
+                        destroyCache("test_jitscc");
+                        Thread.sleep(CLIENT_TEST_TIME_MS);
+
+                        logger.info("Stopping client...");
+                        destroyAndCheckProcess(client, clientBuilder);
+
+                        logger.info("Waiting for " + CLIENT_TEST_TIME_MS + " millis.");
+                        Thread.sleep(CLIENT_TEST_TIME_MS);
+
+                        logger.info("Check whether the cache test_jitscc exists::" + checkCacheExists("test_jitscc"));
```

Comment:
Here we should fail if the cache still exists.

Reply:
Agree. Done the changes.